app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -26,7 +26,7 @@ fi
 
 create_host_curl() {
   curl --request POST <%= foreman_server_url %>/api/hosts \
-       <%= headers.join(' ') %> \
+       <%= '--insecure' if @insecure %> <%= headers.join(' ') %> \
```

Comment:
As this template is actually rendered by Foreman, we can deploy the actual certificate as part of this script and use it to verify this. Imho the main issue is how this template is transferred to a host that should be registered to Foreman. This is where the parameter makes sense.